src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs

```diff
@@ -93,6 +93,7 @@ private static WebHostSettings GetDefaultSettings(ScriptSettingsManager settings
                 settings.ScriptPath = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebJobsScriptRoot);
                 settings.LogPath = Path.Combine(Path.GetTempPath(), @"Functions");
                 settings.SecretsPath = HttpContext.Current.Server.MapPath("~/App_Data/Secrets");
+                settings.IsSelfHost = true;
```

Comment:
@ahmelsayed If you really want to disable all auth for CLI, then you should add a new flag (with big warnings on it) WebHostSettings.DisableAuth.